### PR TITLE
scala3: akka-cluster

### PR DESCRIPTION
**Pull request by octonato**
References #30243

### CODE REVIEW

File: akka-cluster/src/main/java/akka/cluster/protobuf/msg/ClusterMessages.java [hunk 1]
```diff
@@ -15977,23 +15977,23 @@ public int hashCode() {
       }
 
       @java.lang.Override
-      public Builder newBuilderForType() { return newBuilder(); }
-      public static Builder newBuilder() {
+      public Version.Builder newBuilderForType() { return newBuilder(); }
+      public static Version.Builder newBuilder() {
```
Comment (octonato) [inline akka-cluster/src/main/java/akka/cluster/protobuf/msg/ClusterMessages.java:15981]: This is also odd, scala 3 complains of ambiguous reference, but this class is a generated one that we push. 

I fixed it by hand just to pass compilation, but such a solution won't last longer.

Comment (raboof) [reply akka-cluster/src/main/java/akka/cluster/protobuf/msg/ClusterMessages.java:15981]: That sounds like what @ignasi35 ran into for akka-persistence in #30406, https://github.com/lampepfl/dotty/issues/13106

File: akka-cluster/src/main/scala/akka/cluster/Member.scala [hunk 2]
```diff
@@ -345,6 +345,23 @@ final case class UniqueAddress(address: Address, longUid: Long) extends Ordered[
...
+  @nowarn("msg=deprecated")
+  def copy(address: Address) = new UniqueAddress(address, uid.toLong)
+
+  /**
+   * For binary compatibility
+   * Stops `copy(Long)` copy from being generated, use `apply` instead.
+   */
+  @deprecated("Use Long UID constructor instead", since = "2.4.11")
+  @nowarn("msg=deprecated")
+  def copy(uid: Int) = new UniqueAddress(address, uid.toLong)
```
Comment (octonato) [inline akka-cluster/src/main/scala/akka/cluster/Member.scala:364]: Here is an example of the copy method I mentioned in Slack. In the case, it's easy to overload the methods.

We have similar problems with ClusterRouterConfig and ClusterRouterGroupSettings

Comment (raboof) [reply akka-cluster/src/main/scala/akka/cluster/Member.scala:364]: Looks like one option is to convert the case class into a regular class like in https://github.com/akka/akka/pull/29956/commits/cca8f71602438f886cc50b994e35131897be3777 . Painful though.

File: akka-cluster/src/main/java/akka/cluster/protobuf/msg/ClusterMessages.java (hunk 1 above)
Comment (ignasi35) [reply akka-cluster/src/main/java/akka/cluster/protobuf/msg/ClusterMessages.java:15981]: Yeah. It's the exact same cause: inner message on the `.proto` definition (https://github.com/akka/akka/blob/master/akka-cluster/src/main/protobuf/ClusterMessages.proto#L208-L209) causes ambiguous `Builder` reference.

**Comment by octonato** [comment]
@raboof, I rebased this PR with yours #30415. The reason is that cluster tests depends on akka-remote. 

So, #30415 needs to be merged first.

**Review by raboof — APPROVED** [review]

**Comment by octonato** [comment]
This is rebased now.

**Comment by andreaTP** [comment]
LGTM

**Review by andreaTP — APPROVED** [review]

**Comment by raboof** [comment]
```
[info] [JVM-1] LatencySpecMultiJvmNode1:
[info] [JVM-1] Latency of Artery
[info] [JVM-2] LatencySpecMultiJvmNode2:
[info] [JVM-2] Latency of Artery
[info] [JVM-1] - must start echo (on node 'first', class akka.remote.artery.LatencySpecMultiJvmNode1) (52 milliseconds)
[info] [JVM-2] - must start echo (on node 'second', class akka.remote.artery.LatencySpecMultiJvmNode2) (62 milliseconds)
[info] [JVM-1] warmup: 0 msgs/sec total 0 messages
[info] [JVM-1] warmup: 0 msgs/sec total 0 messages
[info] [JVM-1] warmup: 263 msgs/sec, 26,280 bytes/sec, totals 263 messages 0 MB
[info] [JVM-1] warmup: 3,564 msgs/sec, 356,384 bytes/sec, totals 3,829 messages 0 MB
[info] [JVM-1] warmup: 12,333 msgs/sec, 1,233,333 bytes/sec, totals 16,170 messages 1 MB
[info] [JVM-1] Measured send rate 7243 msg/s (new adjustment factor: 1.6567720557779926)
[info] [JVM-1] warmup: 3,711 msgs/sec, 371,121 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] warmup: 0 msgs/sec, 0 bytes/sec, totals 19,884 messages 1 MB
[info] [JVM-1] - must be low for warmup, at 10000 msg/s, payloadSize = 100 (on node 'first', class akka.remote.artery.LatencySpecMultiJvmNode1) *** FAILED *** (22 seconds, 744 milliseconds)
[info] [JVM-1]   java.lang.AssertionError: assertion failed: timeout (22 seconds) during expectMsg: Terminated Actor[akka://LatencySpec/user/$a#1764609543]
[info] [JVM-1]   at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectMsgPF(TestKit.scala:479)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectMsgPF$(TestKit.scala:159)
[info] [JVM-1]   at akka.testkit.TestKit.expectMsgPF(TestKit.scala:973)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectTerminated(TestKit.scala:495)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectTerminated$(TestKit.scala:159)
[info] [JVM-1]   at akka.testkit.TestKit.expectTerminated(TestKit.scala:973)
[info] [JVM-1]   at akka.remote.artery.LatencySpec.test$$anonfun$3$$anonfun$2(LatencySpec.scala:322)
[info] [JVM-1]   at scala.runtime.java8.JFunction1$mcVI$sp.apply(JFunction1$mcVI$sp.scala:18)
[info] [JVM-1]   at scala.collection.immutable.Range.foreach(Range.scala:190)
[info] [JVM-1]   at akka.remote.artery.LatencySpec.test$$anonfun$1(LatencySpec.scala:331)
[info] [JVM-1]   at akka.remote.testkit.MultiNodeSpec.runOn(MultiNodeSpec.scala:373)
[info] [JVM-1]   at akka.remote.artery.LatencySpec.test(LatencySpec.scala:334)
```
:thinking: re-triggering to see if it's flakey

**Pull request merged**